[10.0] IMP l10n_it_rea

eLBati opened the pull request:
FIX REA fields order
ADD REA fields to copany and compute company_registry accordingly

SimoRubi commented on l10n_it_rea/i18n/it.po:
```diff
@@ -1,42 +1,52 @@
 # Translation of Odoo Server.
```
Do not edit po* files

eLBati replied:
Note that we already merged several PRs directly changing po files and no issues were raised. Also, in this case, I checked that no changes were committed by weblate to ```l10n_it_rea/i18n/it.po```
In case of conflicts, I will fix them.

Finally, as previously said, I need to add Italian translation to the module and I can't overload my development workflow

primes2h replied:
In fact you should never push change to .po files.

If you already have a .po file, upload it to weblate.

Here https://docs.google.com/document/d/1b7lzeKyaDJ_zgmZ3VKWNqJyFb5gnsdXF41-DtPKLvRY/edit# i see that

    .pot files are regenerated by Travis and pushed to GitHub as soon as a change is pushed to a main branch in OCA.
...

    Weblate pulls .pot updates from GitHub hourly (at hh:15).
    This can be made instantaneous if you add a push event webhook to your project (https://translation.odoo-community.org/hooks/github)

SimoRubi replied:
@eLBati I do not know if you have been lucky or if editing po* files is actually not harmful.

I don't know how the weblate integration works exactly, @sbidoul surely does, so maybe you should discuss with him if it is possible to edit po* files without risks, and edit https://docs.google.com/document/d/1b7lzeKyaDJ_zgmZ3VKWNqJyFb5gnsdXF41-DtPKLvRY/edit#heading=h.rk8pleq1juou accordingly.

Until then, I will continue to review and prevent po* editing.

eLBati replied:
@sbidoul could you have a look at https://github.com/OCA/l10n-italy/pull/612#discussion_r225777100 and tell me what you think?

sbidoul replied:
@elbati When you push changes to .po files, you risk git merge conflicts with pushes coming from weblate. It will not happen always of course, if nobody is translating at the same time, and there are no .pot file changes, etc. So it's delicate and people really need to understand in detail what they are doing.

When such a merge conflict happens, it's really messy to resolve because you (well, rather, me) currently need shell  access to the weblate server to resolve it (unless you accept losing translations). So yeah, it's messy.

That's why I insist .po files are the responsibility of weblate only.

primes2h replied:
> > Then we are lucky that no-one is translating those modules :)
> 
> Actually @primes2h is.
> 
> @SimoRubi @primes2h see [a07d9c7#diff-bfe1ddd49978694780f98638bfb14cfaR51](https://github.com/OCA/l10n-italy/commit/a07d9c7480eb0fbf7a140dca64846aeaa4abd5a2#diff-bfe1ddd49978694780f98638bfb14cfaR51)
> and
> [167f054#diff-bfe1ddd49978694780f98638bfb14cfaR53](https://github.com/OCA/l10n-italy/commit/167f05402562f2cfaa6687858faec44bd67b2374#diff-bfe1ddd49978694780f98638bfb14cfaR53)
> 
> I changed `l10n_it_fatturapa_out/i18n/it.po` in commit [a07d9c7](https://github.com/OCA/l10n-italy/commit/a07d9c7480eb0fbf7a140dca64846aeaa4abd5a2) , then @primes2h translated that term , using weblate, in commit [167f054](https://github.com/OCA/l10n-italy/commit/167f05402562f2cfaa6687858faec44bd67b2374)
> No issues

Just because your pull request was merged 6 days ago and I modified .po in weblate "after" that (2 days ago). So I made my changes based on your .po.

Think about this situation instead.
You or/and other people make one or more pull request changing .po file, but this PR is merged some days later.
Meanwhile I change the .po file in weblate based on the old .po.
When your PRs will be merged you do get issues.



> 
> > If you already have a .po file, upload it to weblate
> 
> I could follow this way if terms were already present in stable branches, otherwise I should anyway wait for the PR to be merged, then upload .po file to weblate (or translate by web)

primes2h replied:
Just a note.
Please remember to follow wiki guidelines when translating
https://github.com/OCA/l10n-italy/wiki/Traduzioni.

This one http://tp.linux.it/buona_traduzione.html in particular.

eLBati replied:
@sbidoul I see.
If we, l10n-italy contributors, try to not translate at the same time and accept to lose translations in case of conflicts, would the process be less messy and more acceptable?

This would simplify a lot the development process, especially when adding new functionalities, thus new terms (when the term is already available in weblate, we will not change it with a PR).
Otherwise people should wait for the PR to be merged and then translate using weblate, increasing the complexity of the workflow.
We need to make contributors life easy, they are so few...

eLBati replied:
Ok, let's try removing `it.po` from the PR. @primes2h consider that your work, as Italian translation master, could increase, as I expect few people will translate using weblate after merge :sweat_smile:

primes2h replied:
> Ok, let's try removing `it.po` from the PR. @primes2h consider that your work, as Italian translation master, could increase, as I expect few people will translate using weblate after merge sweat_smile

I'll do my best to increase that number, they are so few... :-p

Seriously, this means less work for you before making PR, you can use saved time to help translating/reviewing in weblate later. ;-)
Moreover, I think it can be useful when someone want to test new features in UI.
He can get immediately which and where new added terms are (if other strings have already been translated obviously).

eLBati replied:
@SimoRubi @primes2h https://github.com/OCA/commission/pull/151#discussion_r235674310 :no_mouth:

primes2h replied:
> @SimoRubi @primes2h [OCA/commission#151 (comment)](https://github.com/OCA/commission/pull/151#discussion_r235674310) no_mouth

Per me nessun problema se c'è chiarezza su chi poi si prende in carico la traduzione del modulo, deve essere lo stesso che modifica direttamente i .po tramite la PR.

Sono da evitare le situazioni in cui c'è una persona che traduce/revisiona su weblate e perde tutto il lavoro a causa del merge di una PR fatta da un altro.

eLBati replied:
:+1:

primes2h replied:
> @eLBati When you push changes to .po files, you risk git merge conflicts with pushes coming from weblate. It will not happen always of course, if nobody is translating at the same time, and there are no .pot file changes, etc. So it's delicate and people really need to understand in detail what they are doing.
> 
> When such a merge conflict happens, it's really messy to resolve because you (well, rather, me) currently need shell access to the weblate server to resolve it (unless you accept losing translations). So yeah, it's messy.
> 
> That's why I insist .po files are the responsibility of weblate only.

@eLBati @sbidoul 
I think there could be another potential issue in pushing changes directly to .po files.
I just saw that some it.po files of l10n_italy 12.0 repository modules have the wrong version.
"Project-Id-Version: Odoo Server 10.0\n"
but related .pot file has 12.0.

This is probably caused by pushing changes to it.po manually, forgetting to update that line
(In the actual workflow It should be made automatically by weblate, but it doesn't happen if you push changes to .po).

I don't know if this cause issues in the flow, but it could.

SimoRubi commented on l10n_it_rea/models/res_company.py:
```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Lorenzo Battistini
+
```
Please add license as per https://github.com/OCA/maintainer-tools/blob/9766529146bcb61ce746623da74a23163199bdd3/template/module/models/model_name.py#L3

SimoRubi commented on l10n_it_rea/models/res_company.py:
```diff
@@ -0,0 +1,55 @@
+                        'rea_liquidation_state'
+                    ]._description_selection(self.env)
+                )[company.rea_liquidation_state]
+            company_registry = _("%s %s Share Cap. %s%s %s %s") % (
```
Check `currency.position` to see if the symbol goes before or after the amount, or try to take advantage of a `Monetary` field

eLBati replied:
@SimoRubi do you know in this case how could I take advantage of a Monetary field?

SimoRubi replied:
That would be more nicely displayed in the company form, actually I don't know how that could be useful in string substitution

eLBati commented:
@SimoRubi thanks, I made the other changes

eLBati commented:
@primes2h following https://github.com/OCA/l10n-italy/pull/611#issuecomment-430307355, I moved REA fields to another tab in company form.
I would keep module's scope and name unchanged for now.

@SimoRubi @primes2h do you think this can be merged?

eLBati commented:
@SimoRubi I would keep `rea_capital` a Float because there are other modules using it (as `related`). I fixed the currency symbol to €, because it is an Italian register. I also refactored the computation of `company_registry`, using `onchange`, because in a multinational environment user would need to set `company_registry` manually

primes2h commented:
> @primes2h following [#611 (comment)](https://github.com/OCA/l10n-italy/pull/611#issuecomment-430307355), I moved REA fields to another tab in company form.

Much better. :+1: 

> I would keep module's scope and name unchanged for now.

Ok.

> 
> @SimoRubi @primes2h do you think this can be merged?

Please have a look at #611:

__manifest__.py file
I changed module name to have a uniform naming convention, I saw other modules already use it.
In website field I put a link to github version/module, you point to github repository, what is better?

DESCRIPTION.rst
I made tiny changes in the original text, I also added an Italian description.

If you agree with these changes (mixing with yours if you want) you could include them in v.10.0 too.

SimoRubi commented on l10n_it_rea/models/res_company.py:
```diff
@@ -0,0 +1,55 @@
+                        'rea_liquidation_state'
+                    ]._description_selection(self.env)
+                )[self.rea_liquidation_state]
+            # using always €, as
```
as...?

primes2h commented:
> > In website field I put a link to github version/module, you point to github repository, what is better?
> 
> It is up to you, both are valid: https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11

Ok, I think it would be better to use link of the version/addon.

eLBati commented:
> Ok, I think it would be better to use link of the version/addon

@primes2h done
